atomix/cluster/src/test/java/io/atomix/raft/RaftFailOverTest.java

```diff
@@ -228,7 +228,37 @@ public void shouldReplicateSnapshotAfterDataLoss() throws Exception {
     final var snapshot = raftRule.getSnapshotOnNode(follower);
 
     assertThat(snapshot.getIndex()).isEqualTo(leaderSnapshot.getIndex()).isEqualTo(100);
-    assertThat(snapshot.getTerm()).isEqualTo(snapshot.getTerm());
+    assertThat(snapshot.getTerm()).isEqualTo(leaderSnapshot.getTerm());
+    assertThat(snapshot.getId()).isEqualTo(leaderSnapshot.getId());
+  }
+
+  @Test
+  public void shouldReplicateSnapshotMultipleTimesAfterMultipleDataLoss() throws Exception {
+    // given
+    raftRule.appendEntries(128);
+    raftRule.doSnapshot(100);
+    final var follower = raftRule.shutdownFollower();
+    final var leaderSnapshot = raftRule.getSnapshotFromLeader();
+    raftRule.triggerDataLossOnNode(follower);
+    raftRule.bootstrapNode(follower);
+
+    final var firstSnapshot = raftRule.getSnapshotOnNode(follower);
```

Comment:
Is it guaranteed that the snapshot is available immediately after the bootstrap? Or should it wait until the snapshot is replicated?

Reply:
The test runs without problems so I thought yes. I checked the code again and indeed the follower needs to apply the first entries to become ready.

On bootstrap:

```java
    if (openFutureRef.compareAndSet(null, new AtomixFuture<>())) {
      joiner
          .get()
          .whenComplete(
              (result, error) -> {
                if (error == null) {
                  context.awaitState(
                      RaftContext.State.READY,
                      state -> {
                        started = true;
                        openFutureRef.get().complete(this);
                      });
                } else {
                  openFutureRef.get().completeExceptionally(error);
                }
              });
    }

```

State ready is only set on applying entries https://github.com/zeebe-io/zeebe/blob/develop/atomix/cluster/src/main/java/io/atomix/raft/impl/RaftContext.java#L371-L383